File: 0409. Longest Palindrome/Solution.py
import collections


class Solution:
    def longestPalindrome(self, s: str) -> int:
        ans = 0
        counter = collections.Counter(s)
        for v in counter.values():
            ans += v - (v & 1)
        return ans if ans == len(s) else ans + 1

File: 0409. Longest Palindrome/Solution.java
class Solution {
    public int longestPalindrome(String s) {
        int[] counter = new int[128];
        for (char ch : s.toCharArray())
            counter[ch]++;
        int ret = 0;
        for (int n : counter)
            ret += n - (n & 1);
        return ret < s.length() ? ret + 1 : ret;
    }
}
